File: Binary_Search/704.Binary-Search/704.Binary-Search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low = 0
        high = len(nums) - 1
        if target > nums[high] or target < nums[low]:
            return -1
        while low <= high:
            middle = (low + high) // 2
            if nums[middle] > target:
                high = middle - 1
            elif nums[middle] < target:
                low = middle + 1
            else:
                return middle
        return -1

File: Binary_Search/704.Binary-Search/704.Binary-Search.java
class Solution {
    public int search(int[] nums, int target) {
        int low = 0;
        int high = nums.length - 1;
        if (target > nums[high]) {
            return -1;
        }
        while (low <= high) {
            int middle = (low + high) / 2;
            if (nums[middle] > target) {
                high = middle - 1;
            } else if (nums[middle] < target) {
                low = middle + 1;
            } else {
                return middle;
            }
        }
        return -1;
    }
}
